lib/sources/api/messaging.rb

```diff
@@ -1,16 +1,24 @@
+require "more_core_extensions/core_ext/module/cache_with_timeout"
+
 module Sources
   module Api
     module Messaging
-      def self.client(topic)
+      def self.client
         require "manageiq-messaging"
 
-        @client ||= {}
-        @client[topic] ||= ManageIQ::Messaging::Client.open(
-          :protocol => :Kafka,
-          :host     => ENV["QUEUE_HOST"] || "localhost",
-          :port     => ENV["QUEUE_PORT"] || "9092",
-          :encoding => "json"
-        )
+        Thread.current[:messaging_client] ||= begin
```

Comment:
Went back to caching per thread but want to leave this at 1 puma thread first to test one variable at a time